e2e-polybft/e2e/governance_test.go

```diff
@@ -222,81 +230,58 @@ func TestE2E_Governance_ProposeAndExecuteSimpleProposal(t *testing.T) {
 
-		proposalID := sendProposalTransaction(t, relayer, proposerAcc.Ecdsa,
-			polybftCfg.GovernanceConfig.ChildGovernorAddr,
-			polybftCfg.GovernanceConfig.NetworkParamsAddr,
-			proposalInput, proposalDescription)
-
-		// check that proposal delay finishes, and porposal becomes active (ready to for voting)
-		require.NoError(t, cluster.WaitUntil(3*time.Minute, 2*time.Second, func() bool {
-			proposalState := getProposalState(t, proposalID,
-				polybftCfg.GovernanceConfig.ChildGovernorAddr, relayer)
-
-			return proposalState == Active
-		}))
-
-		// vote for the proposal
-		for _, s := range cluster.Servers {
-			voterAcc, err := helper.GetAccountFromDir(s.DataDir())
-			require.NoError(t, err)
+		executeSuccessfulProposalCycle(t, proposalInput, proposalDescription, "baseFeeChangeDenom", newBaseFeeChangeDenom)
+	})
 
-			sendVoteTransaction(t, proposalID, For, polybftCfg.GovernanceConfig.ChildGovernorAddr,
-				relayer, voterAcc.Ecdsa)
+	t.Run("successful change of block time", func(t *testing.T) {
+		newBlockTime := big.NewInt(5) // 5 seconds
+		setNewBlockTime := contractsapi.SetNewBlockTimeNetworkParamsFn{
+			NewBlockTime: newBlockTime,
 		}
 
-		// check if proposal has quorum (if it was accepted)
-		require.NoError(t, cluster.WaitUntil(3*time.Minute, 2*time.Second, func() bool {
-			proposalState := getProposalState(t, proposalID,
-				polybftCfg.GovernanceConfig.ChildGovernorAddr, relayer)
+		proposalInput, err := setNewBlockTime.EncodeAbi()
+		require.NoError(t, err)
 
-			return proposalState == Succeeded
-		}))
+		proposalDescription := fmt.Sprintf("Change block time from 2s to %ds", newBlockTime)
 
-		// queue proposal for execution
-		sendQueueProposalTransaction(t, relayer, proposerAcc.Ecdsa,
-			polybftCfg.GovernanceConfig.ChildGovernorAddr,
-			polybftCfg.GovernanceConfig.NetworkParamsAddr,
-			proposalInput, proposalDescription)
+		executeSuccessfulProposalCycle(t, proposalInput, proposalDescription, "blockTime", newBlockTime)
 
-		// check if proposal has quorum (if it was accepted)
-		require.NoError(t, cluster.WaitUntil(3*time.Minute, 2*time.Second, func() bool {
-			proposalState := getProposalState(t, proposalID,
-				polybftCfg.GovernanceConfig.ChildGovernorAddr, relayer)
+		// get epoch size from NetworkParams contract
+		networkParamsResponse, err := ABICall(relayer, contractsapi.NetworkParams,
+			polybftCfg.GovernanceConfig.NetworkParamsAddr, types.ZeroAddress, "epochSize")
+		require.NoError(t, err)
 
-			return proposalState == Queued
-		}))
+		epochSize, err := common.ParseUint256orHex(&networkParamsResponse)
+		require.NoError(t, err)
 
 		currentBlockNumber, err := relayer.Client().BlockNumber()
 		require.NoError(t, err)
 
-		// wait for couple of more blocks because of execution delay
-		require.NoError(t, cluster.WaitForBlock(currentBlockNumber+2, 10*time.Second))
+		endOfEpoch := (currentBlockNumber/epochSize.Uint64() + 1) * epochSize.Uint64()
```

Comment:
Maybe currentBlockNumber - 1 (for blocks > 0) should be used instead of currentBlockNumber, otherwise last block in the epoch will be falsely assigned to the next epoch.

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/237/commits/1dbbdcd644edc81de4df36266a9a27ed5d847167